snappy-tools/sbin/collect-debug-artifacts.sh

```diff
@@ -0,0 +1,230 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License. See accompanying
+# LICENSE file.
+#
+
+#!/usr/bin/env bash
+
+usage="Usage: collect-debug-artifacts [ -c|--conf|--config conffile ] [ -h|--help ] \
+       [ -a|--all ] [ -v|--verbose ]"
+
+PWD=`pwd`
+while [ "$1" != "" ]; do
+key="$1"
+
+case $key in
+  -c|--conf|--config)
+  CONF_FILE="$2"
+  shift # past argument
+  ;;
+  -h|--help)
+  echo $usage
+  exit 0
+  ;;
+  -a|--all)
+  GET_EVERYTHING=1
+  ;;
+  -v|--verbose)
+  VERBOSE=1
+  ;;
+esac
+shift # past argument or value
+done
+
+# Check configurations and assign defaults
+function check_configs {
+  if [ -z "${CONF_FILE}" ]; then
+    CONF_FILE="../conf/debug.conf.template"
+  fi
+
+  if [ ! -f "${CONF_FILE}" ]; then
+    echo "Config file ${CONF_FILE} does not exists"
+    exit 1
+  fi
+
+  source $CONF_FILE
+
+  if [ ! -f "${MEMBERS_INFO_FILE}" ]; then
+    echo "MEMBERS_INFO_FILE file ${MEMBERS_INFO_FILE} does not exists"
+    exit 1
+  fi
+
+  if [ -z "${NO_OF_STACK_DUMPS}" ]; then
+    NO_OF_STACK_DUMPS=5
+  fi
+
+  if [ -z "${INTERVAL_BETWEEN_DUMPS}" ]; then
+    INTERVAL_BETWEEN_DUMPS=5
+  fi
+
+  if [ -z "${GET_EVERYTHING}" ]; then
+    GET_EVERYTHING=0
+  fi
+
+  if [ "${VERBOSE}" = "1" ]; then
+    echo CONF=$CONF_FILE
+    echo MEMINFO=$MEMBERS_INFO_FILE
+    echo NUM_STACK_DUMPS=$NO_OF_STACK_DUMPS
+    echo INTERVAL_BETWEEN_DUMPS=$INTERVAL_BETWEEN_DUMPS
+    echo GET_EVERYTHING=${GET_EVERYTHING}
+  fi
+}
+
+function collect_data {
+  host=$1
+  wd=$2
+  srv_num=$3
+
+  if [ "${VERBOSE}" = "1" ]; then
+    echo "Collecting data for process running on ${host} with working_dir ${wd}"
+  fi
+
+  # make a sub directory for this host and pid
+  out_dir="${PWD}/debug_data/${host}-${srv_num}"
+
+  collector_host="pnq-kneeraj3"
+  mkdir -p $out_dir 
+  if [ "${VERBOSE}" = "1" ]; then
+    echo "Args Being passed"
+    echo "arg1 wd = ${wd}" 
+    echo "arg2 num_stack_dumps = ${NO_OF_STACK_DUMPS}" 
+    echo "arg3 interval_dumps = ${INTERVAL_BETWEEN_DUMPS}" 
+    echo "arg4 GET_EVERYTHING = ${GET_EVERYTHING}" 
+    echo "arg5 collector_host = ${collector_host}" 
+    echo "arg6 out dir = ${out_dir}"
+  fi
+
+  typeset -f | ssh $host "$(cat);collect_on_remote ${wd} ${NO_OF_STACK_DUMPS} \\
+      ${INTERVAL_BETWEEN_DUMPS} ${GET_EVERYTHING} ${collector_host} ${out_dir} ${VERBOSE}"
+}
+
+function collect_on_remote {
+  data_dir=$1
+  num_stack_dumps=$2
+  int_stack_dumps=$3
+  get_all=$4
+  collector_host=$5
+  collector_dir=$6
+  verbose=$7
+
+  # first get the pid. The latest log file with the header will have the pid
+  host=`hostname`
+  if [ ! -d $data_dir ]; then
+    echo "${data_dir} not found on host: ${host}"
+  fi
+
+  cd $data_dir
+  files=()
+  last_restart_log=""
+  for l in $( ls -t *.log )
+  do
+    copyright_headers=`grep 'Copyright (C)' ${l}`
```

Comment:
I think there are some variants here in our previous releases. In some the unicode copyright symbol was used instead of `(C)`. Maybe a little more generic pattern: `grep 'Copyright [ ]*([ ]*.[ ]*)'`